test object foo io exception ioexception exception string